sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1305,6 +1328,8 @@ class Analyzer(
             case _: EqualTo | _: EqualNullSafe => false
             case _ => true
           }
+
+          correlated.foreach(checkMixedReferencesInsideAggregation)
```

Comment:
`failOnOuterReference` should be the single entry error checking for outer references. 

`Filter` is special. We can still call `failOnOuterReference(f)`, but do the extra checking for `Filter` in `failOnOuterReference`.